### PR TITLE
Load channels and winpr checks

### PR DESCRIPTION
**Pull request by hardening**
Fix the loading of the rdp2tcp channel.
Adds checks for the kind of object for events and threads.

### CODE REVIEW

File: client/common/cmdline.c [hunk 1]
```diff
@@ -3856,6 +3855,13 @@ BOOL freerdp_client_load_addins(rdpChannels* channels, rdpSettings* settings)
 		}
 	}
 
+	if (settings->RDP2TCPArgs)
+	{
+		if (!freerdp_client_load_static_channel_addin(channels, settings, "rdp2tcp",
```
Comment (akallabeth) [inline client/common/cmdline.c:3860]: Better use the define `RDP2TCP_DVC_CHANNEL_NAME` for the name?

File: client/common/cmdline.c [hunk 2]
```diff
@@ -3629,7 +3629,6 @@ BOOL freerdp_client_load_addins(rdpChannels* channels, rdpSettings* settings)
 		{ FreeRDP_EncomspVirtualChannel, ENCOMSP_SVC_CHANNEL_NAME, settings },
 #endif
 		{ FreeRDP_RemdeskVirtualChannel, REMDESK_SVC_CHANNEL_NAME, settings },
-		{ FreeRDP_RDP2TCPArgs, RDP2TCP_DVC_CHANNEL_NAME, settings->RDP2TCPArgs },
```
Comment (akallabeth) [inline client/common/cmdline.c:3632]: Why not just fix this here? `{ 0, RDP2TCP_DVC_CHANNEL_NAME, settings->RDP2TCPArgs },` should work or not?

File: client/common/cmdline.c (hunk 1 above)
Comment (hardening) [reply client/common/cmdline.c:3860]: Oups good catch

File: client/common/cmdline.c (hunk 2 above)
Comment (hardening) [reply client/common/cmdline.c:3632]: Well if I do this the rdp2tcp channel will always be loaded even with NULL settings->RDP2TCPArgs

**Review by akallabeth — APPROVED** [review]

**Pull request merged**